File: src/main/java/org/openhds/mobile/utilities/StringUtils.java
package org.openhds.mobile.utilities;


public class StringUtils {

    public static String join(String separator, String... strings) {
        if (strings == null)
            return null;
        if (strings.length == 0)
            return "";
        StringBuilder b = new StringBuilder(strings[0]);
        for (int i = 1; i < strings.length; i++) {
            b.append(separator);
            b.append(strings[i]);
        }
        return b.toString();
    }
}

File: src/main/kotlin/org/cimsbioko/utilities/StringUtils.kt
package org.cimsbioko.utilities

import org.cimsbioko.navconfig.UsedByJSConfig

object StringUtils {

    @JvmStatic
    fun join(separator: String, vararg strings: String) = buildString {
        for ((index, str) in strings.withIndex()) {
            append(str)
            if (index < strings.lastIndex) {
                append(separator)
            }
        }
    }

    @UsedByJSConfig
    fun isEmpty(value: String?) = value == null || value.isEmpty()

}